cmd/uid_provider/client/client_test.go

```diff
@@ -722,6 +723,22 @@ func Test_requestNewToken(t *testing.T) {
 	setup()
 	defer teardown()
 
+	// checks whether the two strings are the same or not, handles different
+	// orders
+	check := func(got, want string) bool {
+		gotsplit := strings.Split(got, "-")
+		gotset, _ := utils.MakeStrSet(gotsplit)
+
+		wantsplit := strings.Split(want, "-")
+		wantset, _ := utils.MakeStrSet(wantsplit)
+
+		oldLen := len(wantset)
+		wantset.AppendSet(gotset)
+
+		// if their length stays the same nothing new was added
+		return len(gotset) == len(wantset) && len(gotset) == oldLen
```

Comment:
Set object has a function that returns the disjunction between two sets, if the len of the disjunction isn't 0 the sets were the same. Your implementation also works but DisjunctSet would be a litle simpler. It is up to you though.
```suggestion

		disjunct := utils.DisjunctSet(gotset, wantset)

		// if their length stays the same nothing new was added
		return len(disjunct) == 0
```